File: app/src/main/java/com/dhl/wanandroid/fragment/BaseFragment.java
package com.dhl.wanandroid.fragment;


import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import com.dhl.wanandroid.R;
import com.dhl.wanandroid.util.SettingUtil;
import com.scwang.smartrefresh.layout.api.RefreshLayout;
import com.scwang.smartrefresh.layout.header.ClassicsHeader;
import java.text.SimpleDateFormat;
import java.util.Locale;

/**
 * @author dhl
 * Fragment 基类
 * Fragment 的存在，是 专注于承担 “视图控制器” 责任，
 * 以分担 Activity 责任、让 Activity 更专注于 “幕后协调者” 工作。
 */
public abstract class BaseFragment extends Fragment {

    private static final String TITLE = "title";

    /**
     * 共用ToolBar
     */
    protected Toolbar toolbar;


    /**
     * smartRefresh
     */
    protected RefreshLayout refreshLayout;
    protected ClassicsHeader mClassicsHeader;
    protected Drawable mDrawableProgress;
    /**
     * rcy
     */
    protected RecyclerView recyclerView;




    public BaseFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
    }

    protected void initToolbar(View view) {
        toolbar = view.findViewById(R.id.tool_bar);
        toolbar.setBackground(new ColorDrawable(SettingUtil.INSTANCE.getColor()));
    }

    protected void initRcy(View view) {
        recyclerView = view.findViewById(R.id.rcy_view);
        refreshLayout = view.findViewById(R.id.refreshLayout);
        mClassicsHeader = (ClassicsHeader) refreshLayout.getRefreshHeader();
        //mClassicsHeader.setLastUpdateTime(new Date(System.currentTimeMillis()-deta));
        mClassicsHeader.setTimeFormat(new SimpleDateFormat("更新于 MM-dd HH:mm", Locale.CHINA));
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));

    }

}

File: app/src/main/java/com/dhl/wanandroid/fragment/BaseFragment.kt
package com.dhl.wanandroid.fragment

import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import androidx.paging.LoadState
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.dhl.wanandroid.R
import com.dhl.wanandroid.activity.WebActivity
import com.dhl.wanandroid.adapter.OnItemClickListener
import com.dhl.wanandroid.adapter.WxArticlePgAdapter
import com.dhl.wanandroid.model.Article
import com.dhl.wanandroid.util.SettingUtil.getColor
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch


/**
 * @author dhl
 * Fragment 基类
 * Fragment 的存在，是 专注于承担 “视图控制器” 责任，
 * 以分担 Activity 责任、让 Activity 更专注于 “幕后协调者” 工作。
 */
abstract class BaseFragment : Fragment(), OnItemClickListener {
    /**
     * 共用ToolBar
     */
    protected val toolbar: Toolbar by lazy {
        requireView().findViewById(R.id.tool_bar)
    }

    /**
     * smartRefresh
     */
    protected val refreshLayout: SwipeRefreshLayout by lazy {
        requireView().findViewById(R.id.refreshLayout)
    }


    /**
     * rcy
     */
    protected val recyclerView: RecyclerView by lazy {
        requireView().findViewById(R.id.rcy_view)
    }

    // 使用 lazy 初始化适配器
    protected val basePgAdapter: WxArticlePgAdapter by lazy {
        WxArticlePgAdapter(requireContext(), this)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)

        basePgAdapter.addLoadStateListener { loadState ->
            // 只在加载完成时停止刷新动画
            if (loadState.source.refresh is LoadState.NotLoading) {
                lifecycleScope.launch {
                    delay(500)
                    refreshLayout.isRefreshing = false
                }

            }
        }
    }


    protected fun initToolbar() {
        toolbar.background = (ColorDrawable(getColor()))
    }

    protected fun initRcy() {
        recyclerView.layoutManager = LinearLayoutManager(activity)
    }

    override fun onItemClick(article: Article) {
        WebActivity.startActivity(requireActivity(), article.title, article.link)
    }

    companion object{
        const val ARG_PARAM1 = "param1"
        const val ARG_PARAM2 = "param2"
         inline fun <reified T : Fragment> newInstance(param1: String = "", param2: String = ""): T {
            val fragment = T::class.java.newInstance()
            val args = Bundle()
            args.putString(ARG_PARAM1, param1)
            args.putString(ARG_PARAM2, param2)
            fragment.arguments = args
            return fragment
        }
    }



}